### PR TITLE
BMS-538: Added backoff mechanism when totalTimeout is supplied as setttings

### PR DESCRIPTION
**Pull request by badaldesai**
JIRA Ticket [BMS-538](https://jira.touchtunes.com/browse/BMS-538)

PR is for porting of code from vnext branch to this branch. Here is the PR [93](https://github.com/PlayNetwork/playnetwork-sdk-nodejs/pull/93) for previous changes which is deployed for amfb players. This PR is for core players.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -200,25 +200,37 @@ The supported options are as follows:
 * `content`
   * `host` - the hostname of the content API
   * `secure` - defaults to `true`, defines when the API uses TLS
+  * `totalTimeout` - To activate the expotential backoff mechanism
```
Comment (Mig-OG) [inline readme.md]: I suggest to add unit of measure of timeout ex: seconds or millisecond.

File: lib/request.js [hunk 2]
```diff
@@ -133,6 +137,18 @@ module.exports = (function (self) {
...
+	function isRetry(options, startTime, tryCount) {
+		let retry;
+		if (options.totalTimeout) {
+			const timeElapsed = (new Date()) - startTime;
+			retry = timeElapsed < options.totalTimeout;
+		} else {
+			retry = tryCount <= options.maxRetries;
+		}
+
+		return retry;
```
Comment (Mig-OG) [inline lib/request.js]: May I suggest that form for isRetry function:
```
function isRetry(options, startTime, tryCount) {
    return options.totalTimeout 
        ? (new Date()) - startTime < options.totalTimeout
        : tryCount <= options.maxRetries;
}
```

File: readme.md (hunk 1 above)
Comment (badaldesai) [reply readme.md]: Good suggestion. Added.

File: lib/request.js (hunk 2 above)
Comment (badaldesai) [reply lib/request.js]: Not a fan of shorthand. I feel `if` is just easy and simple to understand. Less strain on eyes and brain :)

**Comment by Mig-OG** [comment]
Do we keep `history.md`?

**Review by Mig-OG — APPROVED** [review]
Look good to me.

**Review by akobyshchatt — APPROVED** [review]

**Pull request merged**